sdk/src/pubkey.rs

```diff
@@ -79,6 +79,21 @@ impl Pubkey {
         ))
     }
 
+    pub fn create_program_address(
+        seeds: &[&str],
+        program_id: &Pubkey,
+    ) -> Result<Pubkey, PubkeyError> {
+        // Use a deterministic base for all program addresses that are owned by `program_id`
+        let mut hash = hashv(&["ProgramDerivedAddress".as_ref(), program_id.as_ref()]);
```

Comment:
Use a Hash object, this creates a new one and closes it every time without additional benefit to security.  Our strings should be last.

Reply:
@aeyakovenko 👀